src/pages/workspaces/List.js

```diff
@@ -499,7 +525,13 @@ export const WorkspaceList = () => {
         workspace: getWorkspace(requestingAccessWorkspaceId),
         onDismiss: () => setRequestingAccessWorkspaceId(undefined)
       }),
-      loadingWorkspaces && (!workspaces ? transparentSpinnerOverlay : topSpinnerOverlay)
+      loadingWorkspaces && (!workspaces ? transparentSpinnerOverlay : topSpinnerOverlay),
+      !loadingWorkspaces && anyPersistentDiskRequiresMigration(workspaces, gcpPersistentDisks) && h(UnboundDiskNotification, {
```

Comment:
It is possible that the workspaces Ajax response has returned, but the persistent disk one has not. I would recommend changing `anyPersistentDiskRequiresMigration` so that it simply returns false if either of `workspaces` or `gcpPersistentDisks` is undefined. Then you can always call it without concern.